File: lib_core/src/main/java/com/lwkandroid/lib/core/java/context/AppContext.java
package com.lwkandroid.lib.core.java.context;

import android.content.Context;

/**
 * Description:全局Context提供者
 *
 * @author LWK
 * @date 2020/1/6
 */
public final class AppContext
{
    private static final Context CONTEXT = ContextProvider.mContext;

    private AppContext() throws IllegalAccessException
    {
        throw new IllegalAccessException("Can't instantiate this class !");
    }

    public static Context get()
    {
        return CONTEXT;
    }
}

File: lib_core/src/main/java/com/lwkandroid/lib/core/kt/AppContext.kt
package com.lwkandroid.lib.core.kt

import android.content.Context
import com.lwkandroid.lib.core.kt.provider.GlobalContextProvider

/**
 * Description:全局Context提供入口，静态内部类单例模式
 * @author LWK
 * @date 2019/12/9
 */
class AppContext private constructor() {

    companion object {
        val instance = InnerHolder.instance
    }

    private object InnerHolder {
        val instance = AppContext()
    }

    private val mContext: Context = GlobalContextProvider.ContextWrapper.context

    fun getContext(): Context {
        return mContext
    }
}
